### DISCUSSION TITLE
Scoring and Probing Service

### DISCUSSION THREAD

**Discussion by jkczyz** [lightningdevkit/rust-lightning#2318] [General]
As part of a Summer of Bitcoin project, we'd like to add a scoring and probing service. It will probe channel balances in the background and update the `ProbabilisticScorer` accordingly. The scoring data then can be served to users in a similar manner as Rapid Gossip Sync (RGS) data.

@007harshmahajan will be working on this project. We'll use this Github discussion for questions that other LDK contributors may help answer.

**Comment by jkczyz** [comment]
One open question that @007harshmahajan and I discussed was where this project should live. It is similar to RGS in that there is a server component responsible for background probing and serving the scoring data. That said, one consideration is whether it should be an independent server or be part of the RGS server.

If independent, the server will need gossip data (i.e., `NetworkGraph`) which it can obtain either as a peer on the P2P gossip network or by downloading it from an RGS server.

If part of an RGS server, then it would simply reuse the `NetworkGraph`.

@TheBlueMatt Do you have a preference here?

**Reply by TheBlueMatt** [reply]
No strong preference, I'd say either in the rust-Lightning repo but separate crate or a separate org repo. I'd say not the RGS server repo, many folks want to use it outside of that context.
